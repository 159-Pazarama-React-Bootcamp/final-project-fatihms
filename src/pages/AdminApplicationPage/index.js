import React, { useEffect } from "react";

import { useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";

import Header from "../../components/Header";
import AdminApplicationForm from "../../containers/AdminApplicationForm";
import Loading from "../../components/Loading";

import { fetchApplications } from "../../redux/ApplicationData/applicationDataSlice";

import styles from "./styles.module.css";

function AdminApplicationPage() {
  const code = useLocation().pathname.split("/")[3];

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchApplications());
  }, [dispatch]);

  const loading = useSelector((state) => state.applicationData.loading);

  /*eslint-disable */
  const foundApplication = useSelector((state) =>
    state.applicationData.applications.find(
      (application) => application.applicationCode === code
    )
  );

  if (loading) {
    return (
      <div className={styles.loading}>
        <Loading />
      </div>
    );
  }
  return (
    <div className={styles["application-status-container"]}>
      <header>
        <Header title="BaÅŸvuru Bilgisi" />
      </header>
      <main>
        <article>
          <AdminApplicationForm foundApplication={foundApplication} />
        </article>
      </main>
    </div>
  );
}

export default AdminApplicationPage;
