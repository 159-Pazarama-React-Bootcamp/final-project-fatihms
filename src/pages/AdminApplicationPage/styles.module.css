.loading {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
  width: 100%;
  background-color: #f5f5f5;
  font-size: 96px;
  font-weight: bold;
  color: darkred;
}

.application-status-container > main {
  width: 100%;
  margin: 60px 0px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.application-status-container > main > article {
  width: 100%;
}
