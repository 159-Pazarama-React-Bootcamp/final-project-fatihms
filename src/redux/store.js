import { configureStore } from "@reduxjs/toolkit";

import codeReducer from "./code/codeSlice";

const store = configureStore({
  reducer: { code: codeReducer },
});

export default store;
