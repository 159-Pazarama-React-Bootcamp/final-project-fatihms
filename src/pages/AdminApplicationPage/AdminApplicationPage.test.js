/*eslint-disable */

import React from "react";

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";

import { Provider } from "react-redux";
import store from "../../redux/store";

import AdminApplicationPage from "./index";

describe("AdminApplicationPage", () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <AdminApplicationPage />
        </Router>
      </Provider>
    );
  });
  test("AdminApplicationPage renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AdminApplicationPage />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
