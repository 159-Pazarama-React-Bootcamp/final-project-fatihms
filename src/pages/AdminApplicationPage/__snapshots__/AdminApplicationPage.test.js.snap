// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminApplicationPage AdminApplicationPage renders correctly 1`] = `
<div
  className="application-status-container"
>
  <header>
    <div
      className="header-container"
    >
      <h2 />
    </div>
  </header>
  <main>
    <article>
      <section
        className="admin-application-form"
      >
        <div
          className="form-container form-container--40"
        >
          <form
            onSubmit={[Function]}
          >
            <h2>
              Başvuru Bilgisi
            </h2>
            <div
              className="aaf-table"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      Başvuru Kodu
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Başvuru Tarihi
                    </td>
                    <td>
                      ss
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Başvuru Yapan Kişi
                    </td>
                    <td>
                      undefined undefined
                    </td>
                  </tr>
                  <tr>
                    <td>
                      TC
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Doğum Tarihi
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Adres
                    </td>
                    <td>
                      undefined undefined undefined
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Başvuru Sebebi
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Durum
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Mesaj
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Diğer
                    </td>
                    <td />
                  </tr>
                  <tr>
                    <td>
                      Durum
                    </td>
                    <td>
                      <input
                        id="status"
                        name="status"
                        onChange={[Function]}
                        placeholder="Durum"
                        type="text"
                        value=""
                      />
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Mesaj
                    </td>
                    <td>
                      <input
                        id="message"
                        name="message"
                        onChange={[Function]}
                        placeholder="Mesaj"
                        type="text"
                        value=""
                      />
                    </td>
                  </tr>
                </tbody>
              </table>
              <button
                className="button-item"
                type="submit"
              >
                Gönder
              </button>
            </div>
          </form>
        </div>
      </section>
    </article>
  </main>
</div>
`;
